### ISSUE TITLE
Prettier wrong formatting of Angular flow

### ISSUE THREAD

**Issue by gultyayev** [prettier/prettier#16222]
**Prettier 3.2.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAltAtJWCYAEAxLvLAQNoBmArgDZ0DOYATnAgLoC8AOiDCxpw+BZiwgMAogDd8jAHw8oBAhmx10jGIuUqCAASoQWBABT4WATwIQqBAO4BDGGAAWGrQEl4AW0amASgIAHwIAazhLaUc6IQBuAgFHMDCCC0sAOgjLIOAlPVVMKCwPGCxXOEcAEzgWHQKVNWK6RwAjODp5YGA02CssyIIAXyHUAHoikraO+r1xydLyypq6pXy9Q2MzRksfVokbO3SM6Ni4BKSU0V39ugydvYkAOUcfOFz1xscAB2+sJxc7k0ZXQvkooLgPm4fAetxEY1mKiGw3W83UwJ0aOKpHwOiUTRw0DIhBIRPwlFoDDE7Cg0P4gmEIFErAkdBkclmBNKiIMRhM5j61lsDmcblK3kh-iCoWyp3iiRYyVSx2yuQM6DsAoEmTlcAydAQAHMYK4PrpGgtgUtqrUeRb1NM6AQwC1GIxeCB0FAtII3rBGFhDeIaN8+F0eirBiMsVN2p1PqoJuitNaVnjzbytqZYQdhcddRdFVcc3cSy83maCqgfn8AWKrRCfODfHSS-CecjO6ik80Mfie4S8LAdCAADQgCDfGBFRjIUCOFjiewABQXCFnKBiTkss-HrSLERgAGVXnAADJeuDIKgxRhwPcHuDH77JL2G5ACITju8+dAfhnjnAAAe3y1OgfowDEAAqtRQAu6BwBuN5MPeICMG+BoAIo0BA8D-l+IAAFaMEBR4YXA2G4VeSDIXe44AI44fAy7iN8G4gI4AZQOwNRVGO9KOOgGhQIaADCEA+D4jjIBxDD8ehIkGgAgjAAjoK0NDMbUF7cdet6oa4MA+HQADqrgQowL5gHAR7rqC6DSKClgyWAbr8dIQieFANSwEerDoFOSneUeMCWAaekoeO3ziHeJmKt8MnRYhtSyPxXp3iwMAsY4hpSRFdEgC+LAZTJjgifQC78dFXowCZ6BVCayAAIwAEwAAzjmwjHoGw2W5dJNH6eOkGtHVDWuMgLXjjQd5QW0SFDSAkLtFUvFnmVho0DlcAAGLGFJqlvqVmkQCAIxAA)

```sh
--parser angular
--print-width 120
--single-quote
```

.prettierrc
```
{
  "trailingComma": "all",
  "bracketSameLine": true,
  "tabWidth": 2,
  "singleQuote": true,
  "semi": true,
  "quoteProps": "as-needed",
  "arrowParens": "avoid",
  "printWidth": 120
}

```

**Input:**

```html
<ion-content #content [fullscreen]="true" scrollEvents>
  <ion-list>
    @for (entry of watchlistItems() | keyvalue; track entry.key) {
    <ion-list-header>
      <ion-label>{{ entry.key }}</ion-label>
    </ion-list-header>

    @for (symbol of entry.value; track symbol.symbolName) {
    <app-watchlist-item [item]="symbol" />
    } }
  </ion-list>
</ion-content>

<ion-content #content [fullscreen]="true" scrollEvents>
  <ion-list>
    @for (entry of watchlistItems() | keyvalue; track entry.key) { @if (entry.value.length) {
    <ion-list-header>
      <ion-label class="instruments-group">{{ entry.key }}</ion-label>
    </ion-list-header>

    @for (symbol of entry.value; track symbol.symbolName) {
    <app-watchlist-item [item]="symbol" />
    } } }
  </ion-list>
</ion-content>

```

**Output:**

```html
<ion-content #content [fullscreen]="true" scrollEvents>
  <ion-list>
    @for (entry of watchlistItems() | keyvalue; track entry.key) {
      <ion-list-header>
        <ion-label>{{ entry.key }}</ion-label>
      </ion-list-header>

      @for (symbol of entry.value; track symbol.symbolName) {
        <app-watchlist-item [item]="symbol" />
      }
    }
  </ion-list>
</ion-content>

<ion-content #content [fullscreen]="true" scrollEvents>
  <ion-list>
    @for (entry of watchlistItems() | keyvalue; track entry.key) {
      @if (entry.value.length) {
        <ion-list-header>
          <ion-label class="instruments-group">{{ entry.key }}</ion-label>
        </ion-list-header>

        @for (symbol of entry.value; track symbol.symbolName) {
          <app-watchlist-item [item]="symbol" />
        }
      }
    }
  </ion-list>
</ion-content>

```

**Expected output:**

The expected output is in the "Output" section of the issue. The "Input" section is what Prettier 3.2.5 actually gives me as a result of `prettier --write`

**Why?**
That's a good question. I cannot reproduce the issue in the playground. However, that's what I get as a result of `prettier --write`. Also, pipelines that run some kind of Linux successfully pass `prettier --check --ignore-unknown .`.

I would have expected, that it may be somehow related to me using MacOS, but then I would have expected the pipelines to fail checks.

`npm list prettier` result
```
├─┬ @qq/blob-src@npm:blob-src@0.0.1 -> ./blob-src
│ ├─┬ @ionic/prettier-config@1.0.1
│ │ └── prettier@2.3.2 deduped
│ ├─┬ prettier-plugin-java@1.0.2
│ │ └── prettier@2.2.1
│ └── prettier@2.3.2
└── prettier@3.2.5
```

In case it's relevant:
- Node v20.11.1
- npm 10.2.4

**Comment by json-derulo** [comment]
It could be that the prettier executable is on an old version. You can verify that by running `prettier --version`. Try a clean npm install.

**Comment by gultyayev** [comment]
It gives 3.2.5.

Fresh install does not help.

**Comment by gultyayev** [comment]
Okay, I had given it a lot of thought and have come to the conclusion, that it was using the wrong parser. After googling a lil' bit, I discovered that it knows that `.html` is Angular only by using the `*.component.html` glob.

The problem was that I was in Ionic project, where there is also a `*.page.html` pattern for pages. Apparently, it was using a standard HTML parser or something of that sorts. By adjusting the config, to use the correct parser for the glob the "bug" was fixed.
